### PR TITLE
Update tags.cncf.io/container-device-interface to v0.7.1

### PR DESCRIPTION
**Pull request by elezar**
This also bumps the maximum supported CDI specification to v0.7.0.

It would also be good to bump this dependency in the 25 release branch.

**- What I did**

Updated the tags.cncf.io/container-device-interface dependency to v0.7.1. This also updates the supported CDI specification to v0.7.0.

**- How I did it**
Bumped the go dependenies and ran `hack/vendor.sh`

**- How to verify it**

Dependencies are bumped.

**- Description for the changelog**

```markdown changelog
Added support for v0.7.0 Container Device Interface (CDI) specifications.
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor.mod [hunk 1]
```diff
@@ -78,7 +78,7 @@ require (
 	github.com/opencontainers/go-digest v1.0.0
 	github.com/opencontainers/image-spec v1.1.0-rc5
 	github.com/opencontainers/runc v1.1.12
-	github.com/opencontainers/runtime-spec v1.1.0
+	github.com/opencontainers/runtime-spec v1.2.0
```
Comment (elezar) [inline vendor.mod:81]: I would assume that this would be a different PR.

Comment (thaJeztah) [reply vendor.mod:81]: Ah, yes, I recall I have this one open, which also contains the update; I can move those commits separately;

- https://github.com/moby/moby/pull/47668

**Comment by thaJeztah** [comment]
@elezar https://github.com/moby/moby/pull/47719 was merged, so you can rebase this one 👍

**Comment by elezar** [comment]
@thaJeztah rebase done.

**Review by vvoland — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!
